File: articles/behavioral-decision-theory/r/run_all_behavioral_decision_workflows.R
# run_all_behavioral_decision_workflows.R
# Run all R workflows for Behavioral Decision Theory.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "behavioral_decision_theory_workflow.R",
  "prospect_theory_profiles.R",
  "probability_weighting_tables.R",
  "framing_sensitivity_reports.R",
  "loss_aversion_review_tables.R",
  "behavioral_diagnostics_summary.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R behavioral decision workflows completed.")

File: articles/behavioral-decision-theory/python/run_all_behavioral_decision_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Behavioral Decision Theory."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "prospect_value_functions.py",
    "probability_weighting_models.py",
    "expected_utility_comparison.py",
    "framing_sensitivity_analysis.py",
    "loss_aversion_diagnostics.py",
    "behavioral_review_queue.py",
    "decision_record_exporter.py",
    "behavioral_decision_theory_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python behavioral decision workflows completed.")


if __name__ == "__main__":
    main()
